src/buffer.hh

```diff
@@ -191,6 +191,9 @@ public:
     // returns true if the buffer is in a different state than
     // the last time it was saved
     bool is_modified() const;
+    // returns true if the buffer is in a different state than
+    // the underlying file
+    bool is_file_modified() const;
```

Comment:
I dont like having that in the Buffer class, we have everything we need to implement that externally, `buffer_utils.cc` would be a good place to have this.

Reply:
I would need getters to get the BOM and EOL format.

Reply:
Those are in the options already